File: python/lld/booking/movie-ticket/show_seat.py
from enums import SeatType, SeatStatus


# Inventory unit. Knows its status + who holds it and until when.
class ShowSeat:
    def __init__(self, seat_id: str, seat_type: SeatType):
        self.id = seat_id
        self.type = seat_type
        self.status = SeatStatus.AVAILABLE
        self.held_by: str | None = None
        self.hold_expires_at = 0.0

    def is_free(self, now: float) -> bool:
        if self.status == SeatStatus.HELD and now > self.hold_expires_at:
            return True
        return self.status == SeatStatus.AVAILABLE

File: java/lld/booking/movie-ticket/ShowSeat.java
// Inventory unit. Knows its status + who holds it and until when.
public class ShowSeat {
    public final String id;
    public final SeatType type;
    public SeatStatus status = SeatStatus.AVAILABLE;
    public String heldBy;
    public long holdExpiresAt;

    public ShowSeat(String id, SeatType type) { this.id = id; this.type = type; }

    public boolean isFree(long now) {
        if (status == SeatStatus.HELD && now > holdExpiresAt) return true;
        return status == SeatStatus.AVAILABLE;
    }
}